src/caret_analyze/architecture/struct/publisher.py

```diff
@@ -70,12 +70,16 @@ def to_value(self) -> PublisherStructValue:
                 else tuple(v.to_value() for v in self.callbacks)),
             construction_order=self.construction_order)
 
-    def assign_callback(self, callback: CallbackStruct):
+    def insert_callback(self, callback: CallbackStruct):
         if self._callbacks is None:
             self._callbacks = [callback]
         elif callback.callback_name not in (self.callback_names or []):
             self._callbacks.append(callback)
 
+    def remove_callback(self, callback: CallbackStruct):
+        if self._callbacks is not None and callback in self._callbacks:
```

Comment:
```suggestion
        if self._callbacks and callback in self._callbacks:
```

Reply:
Fix it in [54ed564](https://github.com/tier4/CARET_analyze/pull/297/commits/54ed56419cac94aaedc2bf635f24f10927e017b8).